File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    stack = []
    for i in range(len(ingredient)):
        stack.append(ingredient[i])
        
        if(len(stack) >= 4):
            l = len(stack)
            if(stack[l - 1] == 1 and stack[l - 2] == 3 and stack[l - 3] == 2 and stack[l - 4] == 1):
                stack.pop()
                stack.pop()
                stack.pop()
                stack.pop()
                answer += 1;
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    var stack = [];
    //pattern 1, 2, 3, 1
    for(let i = 0; i < ingredient.length; i++){
        stack.push(ingredient[i]);
        
        if(stack.length >= 4){
            var len = stack.length;
            if(
                stack[len - 1] == 1 &&
                stack[len - 2] == 3 &&
                stack[len - 3] == 2 &&
                stack[len - 4] == 1
            ){
                stack.pop();
                stack.pop();
                stack.pop();
                stack.pop();
                answer++;
            }
        }
    }
    return answer;
}